File: src/main/java/com/randomperson22/piggymodtest/PiggyModConfig.java
package com.randomperson22.piggymodtest;

import java.io.File;

import net.minecraftforge.common.config.Configuration;

public class PiggyModConfig {
    public static Configuration config;

    public static void init(File configFile) {
        config = new Configuration(configFile);
        syncConfig();
    }

    public static void syncConfig() {

        if (config.hasChanged()) {
            config.save();
        }
    }
}

File: src/main/kotlin/com/randomperson22/piggymodtest/PiggyModConfig.kt
package com.randomperson22.piggymodtest

import net.minecraftforge.common.config.Configuration
import java.io.File

object PiggyModConfig {
    @JvmField
    var config: Configuration? = null

    fun init(configFile: File?) {
        config = Configuration(configFile)
        syncConfig()
    }

    fun syncConfig() {
        if (config!!.hasChanged()) {
            config!!.save()
        }
    }
}